src/Avalonia.Controls/Presenters/ContentPresenter.cs

```diff
@@ -277,6 +277,11 @@ public void UpdateChild()
 
                 Child = newChild;
 
+                if (oldChild?.LogicalParent == this)
```

Comment:
@grokys probably will be even more safe to check for not null here:
`if (oldChild?.LogicalParent != null)`
So we can ensure whatever usage of ContentPresenter is made the child of ContentPresenter will not remain in the logical tree.